gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -422,7 +422,7 @@ private TypeAdapter<Number> floatAdapter(boolean serializeSpecialFloatingPointVa
         }
         float floatValue = value.floatValue();
         checkValidFloatingPoint(floatValue);
-        out.value(value);
+        out.value(floatValue);
```

Comment:
When I ran this change against all of Google's tests, I found two failures, both linked to this line. The problem is that we've only had `JsonWriter.value(float)` since #2130. If people subclassed `JsonWriter` before then and overrode all of the `value` methods, they won't have overridden `value(float)`.

Maybe that kind of subclassing is questionable, I'm not sure. It's certainly rare: we have only 5 subclasses of `JsonWriter`, though 3 of them do override the `value` methods. Still, I think it would safest to undo this change, and then probably also the change in `doubleAdapter` for symmetry.

Reply:
Hmm ok, I think I also need to adjust then `TypeAdapters.FLOAT` as well because that now calls `value(float)` too. To keep the numeric conversion but also remain backward compatible it is now checking that the value is a `Float`, converting it if necessary and calls `value(Number)` again.

I am not changing `doubleAdapter` though.